Sources/LiveViewNative/Shape Modifiers/ShapeModifier.swift

```diff
@@ -0,0 +1,197 @@
+        .init(storage: .falseContent(try component()))
+    }
+}
+
+/// A `ShapeModifier` or `FinalShapeModifier` that switches between two possible shape modifier types.
+struct _ConditionalShapeContent<TrueContent, FalseContent> {
+    enum Storage {
+        case trueContent(TrueContent)
+        case falseContent(FalseContent)
+    }
+    
+    let storage: Storage
+}
+
+extension _ConditionalShapeContent: ShapeModifier
+where TrueContent: ShapeModifier, FalseContent: ShapeModifier {
+    func apply(to shape: some SwiftUI.Shape) -> any SwiftUI.Shape {
+        switch storage {
+        case let .trueContent(modifier):
+            return modifier.apply(to: shape)
+        case let .falseContent(modifier):
+            return modifier.apply(to: shape)
+        }
+    }
+}
+
+extension _ConditionalShapeContent: FinalShapeModifier
+where TrueContent: FinalShapeModifier, FalseContent: FinalShapeModifier {
+    @ViewBuilder
+    func apply(to shape: any SwiftUI.Shape) -> some View {
+        switch storage {
+        case let .trueContent(modifier):
+            modifier.apply(to: shape)
+        case let .falseContent(modifier):
+            modifier.apply(to: shape)
+        }
+    }
+    
+    @ViewBuilder
+    func apply(to shape: any InsettableShape) -> some View {
+        switch storage {
+        case let .trueContent(modifier):
+            modifier.apply(to: shape)
+        case let .falseContent(modifier):
+            modifier.apply(to: shape)
+        }
+    }
+}
+
+/// A protocol used by `ShapeModifierRegistry` to provide access to the `AggregateFinalShapeModifier` return type.
+protocol ShapeModifierRegistryProtocol {
+    associatedtype AggregateFinalShapeModifier: FinalShapeModifier
+    
+    static func decodeShapeModifier(
+        _ type: ShapeModifierType,
+        from decoder: Decoder
+    ) throws -> ShapeModifier
+    
+    @FinalShapeModifierBuilder
+    static func decodeFinalShapeModifier(
+        _ type: FinalShapeModifierType,
+        from decoder: Decoder
+    ) throws -> AggregateFinalShapeModifier
+}
+
+/// Provides helpers for decoding `ShapeModifier` and `FinalShapeModifier`.
+enum ShapeModifierRegistry: ShapeModifierRegistryProtocol {
+    static func decodeShapeModifier(
+        _ type: ShapeModifierType,
+        from decoder: Decoder
+    ) throws -> ShapeModifier {
+        switch type {
+        case .inset:
+            return try InsetModifier(from: decoder)
+        case .offset:
+            return try OffsetShapeModifier(from: decoder)
+        case .rotation:
+            return try RotationModifier(from: decoder)
+        case .size:
+            return try SizeModifier(from: decoder)
+        case .trim:
+            return try TrimModifier(from: decoder)
+        }
+    }
+    
+    @FinalShapeModifierBuilder
+    static func decodeFinalShapeModifier(
+        _ type: FinalShapeModifierType,
+        from decoder: Decoder
+    ) throws -> _ConditionalShapeContent<
```

Comment:
Can this use an opaque type? If other shape modifiers are added, it'd be nice not to have to update the type too.

Reply:
The only thing I can think of is that the conditional conformances of _ConditionalShapeContent to ShapeModifier and FinalShapeModifier depending on the wrapped types might be tripping it up? Either way, not a big deal